worker/changestream/eventmultiplexer/eventmultiplexer.go

```diff
@@ -332,3 +361,23 @@ func (q *EventQueue) gatherSubscriptions(ch changestream.ChangeEvent) []*subscri
 	}
 	return results
 }
+
+// Dispatch fans out the subscription requests against a given term of changes.
+// Each subscription signals the change in a asynchronous fashion, allowing
+// a subscription to not block another change within a given term.
+func (q *EventMultiplexer) dispatchSet(changeSet map[*subscription][]changestream.ChangeEvent) error {
+	w := newNotifyWorker(int64(len(changeSet)))
+
+	// Tie the notify worker to the catacomb, so it will be killed when the
+	// catacomb dies. Thus preventing a non-cancellable mux signalling.
+	if err := q.catacomb.Add(w); err != nil {
```

Comment:
this needs to move to after the Notify calls. As it could cause a panic. Calling tomb.Go is only safe before the tomb is added to a catacomb or inside one of the go routines owned by the tomb.

Alternatively, you could move the Notify logic into the newNotifyWorker.

Reply:
notify logic should be in the notify worker

Reply:
Maybe, I'll have a look into that.